File: AndroidProject/CollectorAgentMind/src/main/java/com/example/collectoragentmind/codelets/sensors/InnerSense.java
package com.example.collectoragentmind.codelets.sensors;

import com.example.collectoragentmind.CollectorAgentMind;

import br.unicamp.cst.core.entities.Codelet;
import br.unicamp.cst.core.entities.MemoryObject;

public class InnerSense extends Codelet {

    /*
        Requests the agent's position every proc. The body
        should respond to this request by updating the
        positionMO through the plugin's interface
    */

    private MemoryObject positionMO;
    private CollectorAgentMind plugin;

    public InnerSense() {
        plugin = CollectorAgentMind.getInstance();
    }

    @Override
    public void accessMemoryObjects() {

    }

    public void proc() {
        plugin.emitRequest("request_agent_position");
    }

    @Override
    public void calculateActivation() {

    }
}

File: GodotKotlin/CoreModelTrail/src/main/kotlin/com/example/game/cst/codelets/sensors/InnerSense.kt
package com.example.game.cst.codelets.sensors

import br.unicamp.cst.core.entities.Codelet
import br.unicamp.cst.core.entities.MemoryObject
import com.example.game.CollectorMindCommunicator

class InnerSense: Codelet() {
    /*
        Requests the agent's position every proc. The body
        should respond to this request by updating the
        positionMO through the plugin's interface
    */

    private var positionMO: MemoryObject? = null

    fun InnerSense() {
    }

    override fun accessMemoryObjects() {
        positionMO = getInput("POSITION") as MemoryObject
    }

    override fun proc() {
        positionMO?.setI(CollectorMindCommunicator.instance.getPosition())
    }

    override fun calculateActivation() {}
}